templates/influxdb.conf.epp

```diff
@@ -12,15 +12,18 @@
       Array $master_list
 | -%>
 
-<% unless $master_list.empty { 
-      $serverarray = $master_list.map |$host| { 
-          $server_tag = join(split($host,"[.]").map | $element | { "server" },'.') 
-          $wildcards = join(split($host,"[.]").map | $element | { "*" },'.') 
-          "puppetlabs.${wildcards}.placeholder .${server_tag}.measurement*" 
-     }
-   }
+<%
+$serverarray = $master_list.map |$host| { 
+  $server_tag = join(split($host,"[.]").map | $element | { "server" },'.') 
+  $wildcards = join(split($host,"[.]").map | $element | { "*" },'.') 
+  "puppetlabs.${wildcards}.placeholder .${server_tag}.measurement*" 
+}
 
-$unique_serverarray = unique($serverarray) 
+if length(delete_undef_values($serverarray)) > 1 {
```

Comment:
This is meaningless if typing the input as suggested above, and you know that the `map` function called above always returns an `Array` because the input to it is known to be `Array[String]`.
The only thing that can happen is that the input can be empty, in which case the `$unique_serverarray` is also empty.

Also, note that the removal of any undef values in `$serverarray` on this line does not alter the `$serverarray`, it just affects the calculated length, so below, the call to `unique` could return a set that incudes `undef` when
there is no protection against that from the start.

You can write the entire thing (lines 15-26) like this:
```puppet
$unique_serverarray = $master_list.map |$host| { 
  $split = $host.split("[.]")
  $server_tag = $split.map | $x | { "server" }.join('.') 
  $wildcards = $split.map | $x | { "*" }.join('.') 
  "puppetlabs.${wildcards}.placeholder .${server_tag}.measurement*" 
}.unique()
```